engine.py

```diff
@@ -295,9 +308,35 @@ def show_panel(self, panel_id, title, bundle, widget_class, *args, **kwargs):
             dock_widget.setFloating(True)
 
         dock_widget.show()
+        # Remember the dock widget, so we can delete it later.
+        self._dock_widgets.append(dock_widget)
 
         return widget_instance
 
+    def close_windows(self):
+        """
+        Closes the various windows (dialogs, panels, etc.) opened by the engine.
+        """
+
+        # Make a copy of the list of Tank dialogs that have been created by the engine and
+        # are still opened since the original list will be updated when each dialog is closed.
+        opened_dialog_list = self.created_qt_dialogs[:]
+
+        # Loop through the list of opened Tank dialogs.
+        for dialog in opened_dialog_list:
+            dialog_window_title = dialog.windowTitle()
+            try:
+                # Close the dialog and let its close callback remove it from the original dialog list.
+                self.log_debug("Closing dialog %s." % dialog_window_title)
+                dialog.close()
+            except Exception, exception:
+                self.log_error("Cannot close dialog %s: %s" % (dialog_window_title, exception))
+
+        # Delete all dock widgets previously added.
+        for dock_widget in self._dock_widgets:
+            MaxPlus.GetQMaxMainWindow().removeDockWidget(dock_widget)
```

Comment:
just double checking - will this work in all versions of max? i think it will because a version that doesn't support this pattern wouldn't have registered a dock widget in the first place, right?